### PR TITLE
FEAT(USD balance showcase) adds USD balance

### LINKED ISSUE #215: [FEAT] Showcase USD with SOL in balance
- We are fetching and displaying SOL in the wallet's balance field, ideally it should be both 
- You can utilize something like Coin Gecko's API to get the rate of Solana in real time

### PR DESCRIPTION
**Pull request by princedotrs**
---

# USD Balance Showcase

Added functionality to show balance in USD along with SOL

**Issue Number:** Fixes #215

## 🛠️ Type of Change

- [ ] 🐛 Bug fix
- [x] ✨ New feature
- [ ] 🚨 Breaking change
- [ ] 📚 Documentation update

## ✅ Checklist

To ensure a smooth review process, please check off each item as you complete it:

- [x] **Code Style:** My code adheres to the project’s style guidelines.
- [x] **Self-Review:** I have reviewed my own code and made improvements.
- [x] **Comments:** I’ve added comments to explain complex or non-obvious parts of the code.
- [ ] **Documentation:** I’ve updated the documentation to reflect my changes.
- [x] **Warnings:** My changes introduce no new warnings.
- [x] **Tests:** I’ve added tests to verify that my changes work as expected.
- [x] **Passes Locally:** All new and existing unit tests pass on my local machine.

----
Screenshot 2024-10-08 at 4 32 01 AM

### CODE REVIEW

File: src/services/walletService.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { pvtKeyEncryptionManager } from '@/actions/pvtKeyEncryptMgmt'
 import base58 from 'bs58'
 
 const customRpcUrl = process.env.NEXT_PUBLIC_SOLANA_RPC || ''
+const coinGeckoUrl = process.env.NEXT_PUBLIC_COIN_GECKO_API || ''
```
Comment (Rudra-Sankha-Sinhamahapatra) [inline src/services/walletService.ts:11]: You can give a default url instead of the null string on  or

**Review by Rudra-Sankha-Sinhamahapatra — COMMENTED** [review]
Nice work

**Review by Rudra-Sankha-Sinhamahapatra — APPROVED** [review]
Everything looks good

File: src/services/walletService.ts (hunk 1 above)
Comment (cb7chaitanya) [reply src/services/walletService.ts:11]: No, let it be keep it an empty string

**Comment by cb7chaitanya** [comment]
@prince981620 Resolve merge conflicts

**Comment by princedotrs** [comment]
hey @cb7chaitanya conflict resolved, review

**Pull request merged**